mapper/tedge_mapper/src/sm_c8y_mapper/mapper.rs

```diff
@@ -313,6 +404,179 @@ impl CumulocitySoftwareManagement {
+    log_content: &str,
+) -> Result<(), SMCumulocityMapperError> {
+    let client = reqwest::ClientBuilder::new().build()?;
+
+    let request = client
+        .post(binary_upload_event_url)
+        .header("Accept", "application/json")
+        .header("Content-Type", "text/plain")
+        .body(log_content.to_string())
+        .bearer_auth(token.token())
+        .timeout(Duration::from_millis(10000))
+        .build()?;
+
+    let _response = client.execute(request).await?;
+    Ok(())
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+/// used to retrieve the id of a log event
+pub struct SmartRestLogEvent {
+    pub id: String,
+}
+
+/// Make a POST request to /event/events and return the event id from response body.
+/// The event id is used to upload the binary.
+async fn create_log_event(
+    url_host: &str,
+    c8y_managed_object: &C8yManagedObject,
+    token: &SmartRestJwtResponse,
+) -> Result<String, SMCumulocityMapperError> {
+    let client = reqwest::ClientBuilder::new().build()?;
+
+    let create_event_url = get_url_for_create_event(&url_host);
+
+    let local: DateTime<Local> = Local::now();
+
+    let c8y_log_event = C8yCreateEvent::new(
+        c8y_managed_object.to_owned(),
+        "c8y_Logfile",
+        &local.format("%Y-%m-%dT%H:%M:%SZ").to_string(),
+        "software-management",
+    );
+
+    let request = client
+        .post(create_event_url)
+        .json(&c8y_log_event)
+        .bearer_auth(token.token())
+        .header("Accept", "application/json")
+        .timeout(Duration::from_millis(10000))
+        .build()?;
+
+    let response = client.execute(request).await?;
+    let event_response_body = response.json::<SmartRestLogEvent>().await?;
+
+    Ok(event_response_body.id)
+}
+
+/// Returns a date time object from a file path or file-path-like string
+/// a typical file stem looks like this: "software-list-2021-10-27T10:29:58Z"
+///
+/// # Examples:
+/// ```
+/// let path_buf = PathBuf::fromStr("/path/to/file/with/date/in/path").unwrap();
+/// let path_bufdate_time = get_datetime_from_file_path(&path_buf).unwrap();
+/// ```
+fn get_datetime_from_file_path(
+    log_path: &PathBuf,
+) -> Result<DateTime<FixedOffset>, SMCumulocityMapperError> {
+    if let Some(stem_string) = log_path.file_stem().and_then(|s| s.to_str()) {
+        // a typical file stem looks like this: software-list-2021-10-27T10:29:58Z.
+        // to extract the date, rsplit string on "-" and take (last) 3
+        let mut stem_string_vec = stem_string.rsplit('-').take(3).collect::<Vec<_>>();
+        // reverse back the order (because of rsplit)
+        stem_string_vec.reverse();
+        // join on '-' to get the date string
+        let date_string = stem_string_vec.join("-");
+        let dt = DateTime::parse_from_rfc3339(&date_string)?;
+
+        return Ok(dt);
+    }
+
+    Err(SMCumulocityMapperError::InvalidDateInFileName(
+        log_path.to_str().unwrap().to_string(),
```

Comment:
This unwrap should go away.
We have a function to convert pathbuf to string with `pathbuf_to_string` from `tedge_utils::paths`